### DISCUSSION TITLE
Custom Value for Excel Style Filtering in Data Grid

### DISCUSSION THREAD

**Discussion by SkylerLutz** [IgniteUI/igniteui-angular#13600] [Q&A] [answered]
Hi, I have a grid with Excel Style Filtering and remote filtering enabled. Most of the data field values are from a small, known set of values, which can easily be pre-fetched and fit into memory, but there is one field that can be any number within +/- 4 billion. I want the user to be able to filter on this column as well as the others, but the excel style filter won't work because it requires that I pre-fetch all discrete values, and they cannot fit in memory. Is there a way to allow the user to specify their own custom value for this field in an excel style filter? That way, the user can get the best of both worlds, meaning, pick from a set of discrete values when there are few and provide their own value when there are too many. Thank you.

**Comment by SkylerLutz** [comment]
@ChronosSF Bump, still seeking an answer, thank you

**Comment by ddaribo** [comment]
Hi @SkylerLutz,

As far as I understand the scenario, I believe it can be achieved by leveraging the [Templates feature](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering#templates) of the Excel Style Filtering component. 
This allows you to modify the content of the [Excel Style Search](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/igxexcelstylesearchcomponent.html) component within it.
This can be conditionally achieved for a column, for example based on its name and/or other condition, such as the number of values being greater than “_n_”.
[This sample](https://stackblitz.com/edit/igx-grid-esf-no-values-for-lengthy-data?file=src%2Fapp%2Fgrid%2Fgrid-excel-style-filtering-load-on-demand%2Fgrid-excel-style-filtering-load-on-demand.component.html) demonstrates an `IgxInputGroup` and two buttons templated within that area to mimic the look of text input search among the ESF list, however, clicking on the “Apply” button results in filtering the grid with the specified value, similar to the quick filtering mode:

![customEsf](https://github.com/IgniteUI/igniteui-angular/assets/35433937/281bb48a-8d77-42ca-bbc2-d86f79f32fe7)

For the purposes of the example, the custom template is set only for the "CompanyName" column.

Please, check out this approach and let me know if it helps achieve your requirement. If this is not an accurate demonstration of what you’re trying to achieve, please, provide more details on the requirement, or feel free to fork and modify the sample as well.

**Comment by SkylerLutz** [comment]
@ddaribo Thank you. Can you please provide another example that accepts and displays multiple search values OR'd together instead of just a single value?  i.e. emulate the behavior of the native excel-style filter, minus the display of all possible search values. Thanks

**Reply by ddaribo** [reply, marked as answer]
Hi @SkylerLutz,

Sure, I can suggest leveraging the [IgxCombo](https://www.infragistics.com/products/ignite-ui-angular/angular/components/combo) at the place of the input. It allows [entering custom values](https://www.infragistics.com/products/ignite-ui-angular/angular/components/combo-features#custom-values)  to the combo’s data list, applying multiple selection and searching among them. 
Similar to the default Excel Style Filtering, when constructing the filtering expressions tree, the values have to be passed as a set to the _searchVal_ argument. Additionally, the filtering condition has to be “_in_”, so the values have to be specified as they are, for example:

![igx-grid-custom-esf](https://github.com/IgniteUI/igniteui-angular/assets/35433937/5e83ce6d-0778-470b-a57b-67d78859429e)


The result set is an “or”-ed filtered list among the specified values for the column.
[Here](https://stackblitz.com/edit/igx-grid-esf-no-values-for-lengthy-data-many-values-or?file=src%2Fapp%2Fgrid%2Fgrid-excel-style-filtering-load-on-demand%2Fgrid-excel-style-filtering-load-on-demand.component.ts) is a link to the new sample I prepared. Please, let me know if I can assist you with anything else on the matter.

**Reply by SkylerLutz** [reply]
Thank you @ddaribo this is almost right. There is a problem that the combo's custom values are accumulated in one list and shown regardless of column. Meaning, I can select Company Names that I have added even if I am filtering by Employees. Is there a way to separate the custom values per column? Thank you

**Reply by ddaribo** [reply]
Hi @SkylerLutz,

I am glad that you find the suggested approach helpful!
And sure, [here](https://stackblitz.com/edit/igx-grid-esf-no-values-for-lengthy-data-many-values-or-2?file=src%2Fapp%2Fgrid%2Fgrid-excel-style-filtering-load-on-demand%2Fgrid-excel-style-filtering-load-on-demand.component.ts) is a fork of the previous sample, where the custom filtering combo is introduced for the Excel Style Filtering UI of the “ID” column as well.
